integration_test/test/code_generation/app_with_defaults_test.exs

```diff
@@ -1,12 +1,314 @@
 defmodule Phoenix.Integration.CodeGeneration.AppWithDefaultsTest do
   use Phoenix.Integration.CodeGeneratorCase, async: true
 
-  test "newly generated app has no warnings or errors" do
-    with_installer_tmp("new with defaults", fn tmp_dir ->
-      app_root_path = generate_phoenix_app(tmp_dir, "phx_blog")
+  describe "new with defaults" do
+    test "has no compilation or formatter warnings" do
+      with_installer_tmp("new with defaults", fn tmp_dir ->
+        {app_root_path, output} = generate_phoenix_app(tmp_dir, "phx_blog")
 
-      assert_no_compilation_warnings(app_root_path)
-      assert_passes_formatter_check(app_root_path)
-    end)
+        assert_file(Path.join(app_root_path, "README.md"))
+
+        assert_file(Path.join(app_root_path, ".formatter.exs"), fn file ->
```

Comment:
I would honestly not move these assertions here. They are perfect for unit testing. Making sure the code compiles (and is formatted) and so on are a great use for integration tests tho!

Reply:
Perfect. I thought they might be faster and more appropriate as unit tests too. 👍I did this because I thought there might have been a desire to remove `phx_new_test.exs`, but I'm thinking it makes sense to leave it.

Do you think we should have the code compilation and formatter checks in separate tests from running a database-backed test suite? It will be slower, but I'm thinking this may still be a good idea, so people can still get some benefit from the test suite if the databases aren't running.

Reply:
We want to remove `phx_new_test` but the one inside "test/mix/tasks", not the installer ones. :)

> Do you think we should have the code compilation and formatter checks in separate tests from running a database-backed test suite?

I would say so because we want to check the `--no` options and umbrella stuff works too. I would reserve the database tests to check the output of the other complex `mix phx.gen.*` generators most of all. I guess one way to see this is that the database connection is required to test `mix phx.gen.*` more than anything else.